Server/config/passport.config.js

```diff
@@ -1,13 +1,49 @@
+const passport = require('passport');
+var LocalStrategy = require('passport-local').Strategy;
+const passportJWT = require('passport-jwt');
+const JWTStrategy = passportJWT.Strategy;
+const ExtractJwt = require('passport-jwt').ExtractJwt;
+const bcrypt = require('bcrypt');
+const userModel = require('../models/user.model');
+const saltRounds = 10;
 
-/*
- * @class passportConfig
- */
-class passportConfig {
-
-    /**
-    * Initializes the passport config
-    */
+// Local passport strategy is used to authenticating user and generating a token  
+passport.use(new LocalStrategy({
+usernameField: 'username',
+passwordField: 'password',
+}, async (username, password, done) => {
+try {
+    var userDocument = await userModel.findOne({username: username}).exec();
```

Comment:
we can get only the password field from db instead of complete userData

Reply:
We need to return userdocument after comparing the passwords.
Please suggest if it is fine.